src/test/ui/or-patterns/feature-gate-const-fn.rs

```diff
@@ -31,6 +30,8 @@ fn main() {
         let x = Ok(3);
         let Ok(y) | Err(y) = x;
         //~^ ERROR or-pattern is not allowed in a `const`
+        //~| ERROR constant contains unimplemented expression type
```

Comment:
What is going on here? cc @ecstatic-morse @oli-obk

Reply:
We're generating actual branches in the MIR now, which const qualif is erroring for.

Reply:
What surprised me though was that we've added support for control flow to const qualif, so I didn't expect this specific error here. Should we open an issue to follow up?

Reply:
You're right, this is from the  `ProjectionElem::Downcast`. Changing from `y` to `_` avoids the extra errors.